client/my-sites/earn/memberships/index.jsx

```diff
@@ -577,27 +587,24 @@ class MembershipsSection extends Component {
 const mapStateToProps = ( state ) => {
 	const site = getSelectedSite( state );
 	const siteId = getSelectedSiteId( state );
+	const earnings = getEarningsWithDefaultsForSiteId( state, siteId );
 
 	return {
 		site,
 		siteId,
 		siteSlug: getSelectedSiteSlug( state ),
-		total: get( state, [ 'memberships', 'earnings', 'summary', siteId, 'total' ], 0 ),
-		lastMonth: get( state, [ 'memberships', 'earnings', 'summary', siteId, 'last_month' ], 0 ),
-		forecast: get( state, [ 'memberships', 'earnings', 'summary', siteId, 'forecast' ], 0 ),
-		currency: get( state, [ 'memberships', 'earnings', 'summary', siteId, 'currency' ], 'USD' ),
-		commission: get( state, [ 'memberships', 'earnings', 'summary', siteId, 'commission' ], '0.1' ),
-		totalSubscribers: get( state, [ 'memberships', 'subscribers', 'list', siteId, 'total' ], 0 ),
-		subscribers: get( state, [ 'memberships', 'subscribers', 'list', siteId, 'ownerships' ], {} ),
-		connectedAccountId: get(
-			state,
-			[ 'memberships', 'settings', siteId, 'connectedAccountId' ],
-			null
-		),
-		connectUrl: get( state, [ 'memberships', 'settings', siteId, 'connectUrl' ], '' ),
+		total: earnings.total,
+		lastMonth: earnings.last_month,
+		forecast: earnings.forecast,
+		currency: earnings.currency,
+		commission: earnings.commission,
+		totalSubscribers: getTotalSubscribersForSiteId( state, siteId ),
+		subscribers: getOwnershipsForSiteId( state, siteId ) ?? {},
```

Comment:
nit: move the default `{}` to the selector for consistency

Reply:
Also, all these default object/array literals will cause endless rerenders in case if the selector result is `null`. Another reason to move it inside the selector.

Reply:
Moved to the selector, it is indeed better in this case. As for avoiding rerenders, that could be done in the component module as well, which is what I'm doing for the other selector below.

Reply:
> As for avoiding rerenders, that could be done in the component module as well

Proper memoization, though, is what people usually expect from selectors themselves. The component shouldn't need to optimize that. It should be OK to just use `connect` in the most simple form.

Reply:
Yes, absolutely; that's definitely the best practice 👍